gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/SimpleDescriptor.java

```diff
@@ -31,15 +34,19 @@
 
     String getDiscoveryPasswordAlias();
 
-    String getClusterName();
+    String getCluster();
 
     String getProviderConfig();
 
     List<Service> getServices();
 
+    Service getService(String serviceName);
+
     List<Application> getApplications();
 
+    Application getApplication(String applicationName);
```

Comment:
What are Applications here?

Reply:
I'm not sure if I understand your question. This is a new simple method to return an application - identified by the application name - and not the entire collection.